### PR TITLE
Petres changes

### PR DESCRIPTION
**Pull request by joshking**
Ticket: https://saucedev.atlassian.net/browse/GROW-123

Visual improvements including: 

- CSS changes
- Slight DOM changes (within html, and js files)
- Visual improvements
- Added fat sticky footer

Screenshots in ticket.

### CODE REVIEW

**Comment by richard-sl** [comment]
Current state of tests seems to be:
```
Test Suites:     0 passed, 6 failed, 6 total (100% completed)
```

The 6 failing are failing because of sessionStorage. I suspect `localhost` is being special in comparison to `127.0.0.1`.

I will try switching tests to `127.0.0.1` from `localhost` and see if that satisfies tests.

File: src/checkout-complete.html [hunk 1]
```diff
@@ -7,21 +7,34 @@
...
   </div>
+  <footer class="footer">
+    
+      <ul class="social">
+        <li class="social_twitter">Twitter</li>
+        <li class="social_facebook">Facebook</li>
+        <li class="social_linkedin">LinkedIn</li>
+      </ul>
+     
+      <div class="footer_copy">&copy; 2018 Sauce Labs. All Rights Reserved.  Terms of Service | Privacy Policy</div>
```
Comment (richard-sl) [inline src/checkout-complete.html]: 2019?

File: src/js/components/login-controls.js [hunk 2]
```diff
@@ -66,9 +66,9 @@ class LoginButton extends Component {
 	  return (
 		<div className="login-box">
 		  <form onSubmit={this.handleSubmit}>
-			  <input type="text" className="login-input" data-test="username" id="user-name" placeholder="Username" value={this.state.username} onChange={this.handleUserChange} />			  
-			  <input type="password" className="login-input" data-test="password" id="password" placeholder="Password" value={this.state.password} onChange={this.handlePassChange} />			  
-			  <input type="submit" className="login-button" value="LOGIN" />
+			  <input type="text" className="form_input" data-test="username" id="user-name" placeholder="Username" value={this.state.username} onChange={this.handleUserChange} />			  
+			  <input type="text" className="form_input" data-test="password" id="password" placeholder="Password" value={this.state.password} onChange={this.handlePassChange} />
```
Comment (richard-sl) [inline src/js/components/login-controls.js]: Why the change from `password` type to `text` on a password form input? 

Also, why the className change? That seems like it _could_ break tests for anyone not going by the better ID. I don't have a strong opinion here just trying to understand.

Comment (joshking) [reply src/js/components/login-controls.js]: I thought this might come up. I should have mentioned it. I'm not concerned about the password being stolen or observed so I don't think we need that little layer of obfuscation. I also found that as I was typing the password in, I didn't know if I had entered it in correctly. I err'd on the side of the user not being hindered at all during the login process. If it's a problem let me know. 

The form_input class is a reusable, and standard class name for objects like form inputs. Adding a class for individual elements isn't the best practice unless they are drastically different. Helps keep us from contracting "Class-itis." I tried to standardize all of the buttons in the app. For instance "remove-from-cart" button is actually just the secondary button style, and can be referenced by .btn-secondary along with the back button in the cart. In other words, individual elements should try to take advantage of more global styles, and specifically targeting individual elements that are reusable (buttons, form inputs) might not be the best approach. I would also recommend using ID's vs classes for tests if possible.

Comment (richard-sl) [reply src/js/components/login-controls.js]: I think the only problem with not having input type `password` is that we'd be deliberately doing something a user should _almost without exception_ never do. 

I do think that the current username/password combinations are annoying on keyboard and especially mobile keyboard because `_` is far less accessible than `-`; and the separator is used in both username and password. I'd think changing the password to address the typing password problem might be more desirable? (of course that'd highly likely break downstream CI - but better sooner than later)

Comment (joshking) [reply src/js/components/login-controls.js]: Sounds like you'd prefer to leave it as password. Not a problem at all! I'll swap this back.

**Comment by richard-sl** [comment]
Currently failing test:
```
[chrome #0-5] Spec: /usr/local/saucelabs/sample-app-web/test/e2e/specs/slider.menu.spec.js
[chrome #0-5] Running: chrome
[chrome #0-5]
[chrome #0-5] Slider Menu
[chrome #0-5]    âœ“ should be able to open and close the slider menu
[chrome #0-5]    âœ“ should be able to navigate to menu about page
[chrome #0-5]    âœ“ should be able to log out via menu
[chrome #0-5]    âœ“ should be able to navigate to menu inventory page
[chrome #0-5]    âœ– should clear the cart on reset app state
[chrome #0-5]
[chrome #0-5] 4 passing (9.9s)
[chrome #0-5] 1 failing
[chrome #0-5]
[chrome #0-5] 1) Slider Menu should clear the cart on reset app state
[chrome #0-5] TypeError: Cannot read property 'click' of undefined
[chrome #0-5]     at <Jasmine>
[chrome #0-5]     at UserContext.it (/usr/local/saucelabs/sample-app-web/test/e2e/specs/slider.menu.spec.js:128:35)
[chrome #0-5]     at <Jasmine>
```
```
[chrome #0-3] Spec: /usr/local/saucelabs/sample-app-web/test/e2e/specs/inventory.list.spec.js
[chrome #0-3] Running: chrome
[chrome #0-3]
[chrome #0-3] Inventory List
[chrome #0-3]    âœ“ should land on inventory list page after login
[chrome #0-3]    âœ“ should be able to open and close the slider menu
[chrome #0-3]    âœ“ should be able to navigate to shopping cart page
[chrome #0-3]    âœ– should be able to add items to cart and remove items from cart
[chrome #0-3]    âœ“ should be able to navigate to item page via item title
[chrome #0-3]    âœ“ should be able to navigate to item page via item image
[chrome #0-3]    âœ“ item images should not be broken
[chrome #0-3]    âœ“ sort should readjust item list order
[chrome #0-3]
[chrome #0-3] 7 passing (9.3s)
[chrome #0-3] 1 failing
[chrome #0-3]
[chrome #0-3] 1) Inventory List should be able to add items to cart and remove items from cart
[chrome #0-3] TypeError: Cannot read property 'click' of undefined
[chrome #0-3]     at <Jasmine>
[chrome #0-3]     at UserContext.it (/usr/local/saucelabs/sample-app-web/test/e2e/specs/inventory.list.spec.js:79:35)
[chrome #0-3]     at <Jasmine>
```
```

[chrome #0-2] Spec: /usr/local/saucelabs/sample-app-web/test/e2e/specs/inventory.item.spec.js
[chrome #0-2] Running: chrome
[chrome #0-2]
[chrome #0-2] Inventory Item
[chrome #0-2]    âœ“ should be able to open and close the item page slider menu
[chrome #0-2]    âœ“ should be able to navigate to item page shopping cart page
[chrome #0-2]    âœ– should be able to add items to cart and remove items from cart
[chrome #0-2]    âœ“ should be able to navigate back to item list page from item page
[chrome #0-2]    âœ“ should have the same title on the list and item pages
[chrome #0-2]
[chrome #0-2] 4 passing (20.5s)
[chrome #0-2] 1 failing
[chrome #0-2]
[chrome #0-2] 1) Inventory Item should be able to add items to cart and remove items from cart
[chrome #0-2] Error: Can't call click on element with selector ".add-to-cart-button" because element wasn't found
[chrome #0-2] Error: Can't call click on element with selector ".add-to-cart-button" because element wasn't found
```
```
[chrome #0-1] Spec: /usr/local/saucelabs/sample-app-web/test/e2e/specs/checkout.info.spec.js
[chrome #0-1] Running: chrome
[chrome #0-1]
[chrome #0-1] Checkout Info
[chrome #0-1]    âœ– should be able to open and close the slider menu
[chrome #0-1]    âœ– should be able to navigate to shopping cart page
[chrome #0-1]    âœ– should be able to return to cart page from checkout info page
[chrome #0-1]    âœ– should be able to continue checkout process from info page
[chrome #0-1]
[chrome #0-1] 4 failing (35.1s)
[chrome #0-1]
[chrome #0-1] 1) Checkout Info should be able to open and close the slider menu
[chrome #0-1] Error: Can't call click on element with selector ".cart_checkout_link" because element wasn't found
[chrome #0-1] Error: Can't call click on element with selector ".cart_checkout_link" because element wasn't found
[chrome #0-1]
[chrome #0-1] 2) Checkout Info should be able to navigate to shopping cart page
[chrome #0-1] Error: Can't call click on element with selector ".cart_checkout_link" because element wasn't found
[chrome #0-1] Error: Can't call click on element with selector ".cart_checkout_link" because element wasn't found
[chrome #0-1]
[chrome #0-1] 3) Checkout Info should be able to return to cart page from checkout info page
[chrome #0-1] Error: Can't call click on element with selector ".cart_checkout_link" because element wasn't found
[chrome #0-1] Error: Can't call click on element with selector ".cart_checkout_link" because element wasn't found
[chrome #0-1]
[chrome #0-1] 4) Checkout Info should be able to continue checkout process from info page
[chrome #0-1] TypeError: Cannot read property 'click' of undefined
[chrome #0-1]     at <Jasmine>
[chrome #0-1]     at UserContext.it (/usr/local/saucelabs/sample-app-web/test/e2e/specs/checkout.info.spec.js:118:35)
[chrome #0-1]     at <Jasmine>
```
```
[chrome #0-0] Spec: /usr/local/saucelabs/sample-app-web/test/e2e/specs/cart.spec.js
[chrome #0-0] Running: chrome
[chrome #0-0]
[chrome #0-0] Shopping Cart
[chrome #0-0]    âœ“ should be able to open and close the cart page slider menu
[chrome #0-0]    âœ“ should be able to navigate to cart page shopping cart page
[chrome #0-0]    âœ– should be able to add items to cart and see on cart page
[chrome #0-0]    âœ– should be able to remove items from cart on cart page
[chrome #0-0]    âœ– should be able to return to inventory list from cart page
[chrome #0-0]    âœ– should be able to start checkout process from cart page
[chrome #0-0]
[chrome #0-0] 2 passing (17.1s)
[chrome #0-0] 4 failing
[chrome #0-0]
[chrome #0-0] 1) Shopping Cart should be able to add items to cart and see on cart page
[chrome #0-0] TypeError: Cannot read property 'click' of undefined
[chrome #0-0]     at <Jasmine>
[chrome #0-0]     at UserContext.it (/usr/local/saucelabs/sample-app-web/test/e2e/specs/cart.spec.js:83:35)
[chrome #0-0]     at <Jasmine>
[chrome #0-0]
[chrome #0-0] 2) Shopping Cart should be able to remove items from cart on cart page
[chrome #0-0] TypeError: Cannot read property 'click' of undefined
[chrome #0-0]     at <Jasmine>
[chrome #0-0]     at UserContext.it (/usr/local/saucelabs/sample-app-web/test/e2e/specs/cart.spec.js:117:35)
[chrome #0-0]     at <Jasmine>
[chrome #0-0]
[chrome #0-0] 3) Shopping Cart should be able to return to inventory list from cart page
[chrome #0-0] Error: Can't call click on element with selector ".cart_cancel_link" because element wasn't found
[chrome #0-0] Error: Can't call click on element with selector ".cart_cancel_link" because element wasn't found
[chrome #0-0]
[chrome #0-0] 4) Shopping Cart should be able to start checkout process from cart page
[chrome #0-0] TypeError: Cannot read property 'click' of undefined
[chrome #0-0]     at <Jasmine>
[chrome #0-0]     at UserContext.it (/usr/local/saucelabs/sample-app-web/test/e2e/specs/cart.spec.js:183:35)
[chrome #0-0]     at <Jasmine>
```

File: src/checkout-complete.html (hunk 1 above)
Comment (joshking) [reply src/checkout-complete.html]: I can fix this a couple ways. The first would be to dynamically display with vanilla js: <script>document.write(new Date().getFullYear())</script>

or we could write that in the react side, and I'd feel better if Ryan put that in. 

or we could just hard code it as 2019 and come back to in 2020 after someone notices :)

Comment (richard-sl) [reply src/checkout-complete.html]: The latter is probably good enough for now - just make sure there is a story on the backlog for the dynamic solution.
